front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -424,7 +438,7 @@ export default function AssistantBuilder({
           setAssistantHandleError("Only letters, numbers, _ and - allowed");
         }
         valid = false;
-      } else if (!assistantHandleIsAvailable(builderState.handle)) {
+      } else if (!(await assistantHandleIsAvailable(builderState.handle))) {
```

Comment:
This is called even if I update the description. Let's refactor a bit so that it's called only if the name is being changed?